### ISSUE TITLE
[ARM] Cannot use multiple addresses for firewall rules

### ISSUE THREAD

**Issue by tanasegabriel** [digitalocean/terraform-provider-digitalocean#364]
### Terraform Version
```
# terraform -version
Terraform v0.12.20
+ provider.digitalocean v1.13.0
```

### Affected Resource(s)
Please list the resources as a list, for example:
- `digitalocean_firewall`

### Terraform Configuration Files
```hcl
resource "digitalocean_firewall" "steam_server" {
  name = "steam-server"

  droplet_ids = [digitalocean_droplet.steam_server.id]

  inbound_rule {
    protocol         = "tcp"
    port_range       = "27015"
    source_addresses = ["0.0.0.0/0", "::/0" ]
  }
}
```

### Debug Output
https://gist.github.com/tanasegabriel/616ee91d17a97ab0698936b0f26fdd92

### Panic Output
N/A

### Expected Behavior
The provider should successfully parse the items of  `destination_addresses` and/or `source_addresses` and create a firewall rule with multiple destionations / sources.

### Actual Behavior
The provider crashes whenever the number of items defined in the list is greater than 1:

```
------------------------------------------------------------------------

Error: rpc error: code = Unavailable desc = transport is closing


panic: missing field in set: inbound_rule.216503885.source_addresses.217649824.217649824

(...) - see the full output in the attached gist

2020/02/01 10:57:55 [TRACE] dag/walk: upstream of "provider.digitalocean (close)" errored, so skipping
2020/02/01 10:57:55 [TRACE] dag/walk: upstream of "meta.count-boundary (EachMode fixup)" errored, so skipping
2020/02/01 10:57:55 [TRACE] dag/walk: upstream of "root" errored, so skipping
2020/02/01 10:57:55 [INFO] backend/local: plan operation completed
2020/02/01 10:57:55 [TRACE] statemgr.Filesystem: removing lock metadata file .terraform.tfstate.lock.info
2020/02/01 10:57:55 [TRACE] statemgr.Filesystem: unlocking terraform.tfstate using fcntl flock
2020-02-01T10:57:55.043Z [DEBUG] plugin: plugin exited
```

### Steps to Reproduce
1. On an ARM device, running `terraform_0.12.20_linux_arm`, define a DigitalOcean firewall with an inbound rule (or an outbound one) containing multiple address.
2. `terraform plan`

### Important Factoids
I was able to reproduce this only with ARM architecture, on a native device (Raspberry Pi 4), as well as on a x64 machine running QEMU emulation.
The more commonly used amd64 architecture does not present this issue.

A simple workaround is to define multiple rules for the same protocol and ports, e,g,:

```hcl
resource "digitalocean_firewall" "steam_server" {
  name = "steam-server"

  droplet_ids = [digitalocean_droplet.steam_server.id]

  inbound_rule {
    protocol         = "tcp"
    port_range       = "27015"
    source_addresses = ["0.0.0.0/0"]
  }

  inbound_rule {
    protocol         = "tcp"
    port_range       = "27015"
    source_addresses = ["::/0"]
  }
}
```
This has the potential of becoming "messy".

**Comment by eddiezane** [comment]
@tanasegabriel apologies that you ran into an issue here and thanks for reporting.

I am indeed able to reproduce this on my Raspberry Pi but like you said works fine on x64.

I think this might be a bug in Terraform itself. I'll ping some folks on Monday and we'll probably need to get this opened on the main repo.

**Comment by paultyng** [comment]
@tanasegabriel what is the value of `digitalocean_droplet.steam_server.id`?

**Comment by tanasegabriel** [comment]
@paultyng you're looking for a specific example here? I use Terraform to spin up a bunch of DigitalOcean resources when required, followed by destroying them when they're not in use.

The ID changes with each of these cycles, as expected. The last known example would be `180305130`, though I'm not sure how this helps. The droplet associated with that ID does not exist anymore, nor do I have any droplet running 24/7.

That ID is "outputted" by the standard `digitalocean_droplet` resource block - I'm assuming it will have the same format every time.

**Comment by tanasegabriel** [comment]
This is still occurring with Terraform 0.12.23 for ARM and `provider.digitalocean v1.14.0`

**Comment by davidcollom** [comment]
I'm also seeing this issue on ARM. Terraform version: `0.12.24` and Provider version: `v1.15.1`

**Comment by tanasegabriel** [comment]
This seems to have been fixed sometime between  `v1.22.2` and `v2.7.0`. Just ran a few upgrades and everything seems to be working as expected. `terraform version` for future reference:

```
Terraform v0.14.10
+ provider registry.terraform.io/digitalocean/digitalocean v2.7.0
+ provider registry.terraform.io/hashicorp/null v3.1.0
+ provider registry.terraform.io/hashicorp/template v2.2.0
```

**Comment by andrewsomething** [comment]
Thanks for following up!
